api.js

```diff
@@ -777,6 +777,28 @@ API.prototype.recordCustomEvent = function recordCustomEvent(eventType, attribut
   this.agent.customEvents.add([instrinics, attributes])
 }
 
+/**
+ * Shuts down the NR agent
+ * @param {object} [options=] - object with shut down options
+ * @param {boolean} [options.collectPendingData=] - boolean value that tells if NR agent must forsibly send pending data
+ * @param {function} [callback=new Function()]  - callback function, that runs when agent stopped
+ */
+API.prototype.shutdown = function shutdown(options, cb) {
+    var metric = this.agent.metrics.getOrCreateMetric(
+        NAMES.SUPPORTABILITY.API + '/shutdown'
+    );
+    metric.incrementCallCount();
+    var callback = cb || new Function();
```

Comment:
If no callback is provided, log and return - see createTracer() for an example.  Also, options is optional, so maybe check if only callback was provided.